CI: Flaky test on Windows

Surprisingly enough, the same test is failing randomly on Windows CI. An example of a failing job: https://github.com/strictdoc-project/strictdoc/actions/runs/6683385940/job/18159561194?pr=1395.

```
--- Received input ---



WARNING:  Current configuration will not reload as not all conditions are met, please refer to documentation.


INFO:     Started server process [3436]


INFO:     Waiting for application startup.


INFO:     Application startup complete.


ERROR:    [Errno 10013] error while attempting to bind on address ('::1', 51000, 0, 0): an attempt was made to access a socket in a way forbidden by its access permissions


INFO:     Waiting for application shutdown.


INFO:     Application shutdown complete.

tests\end2end\project_options\project_title\02_option_specified\test_02_option_specified.py -> Test 23/187

TestSDocServer: Server is up and running on port: 5112.
                        current_expectation = expectations[0]

                        received_input.append(line_string)

                        if line_string.rstrip() in current_expectation:

                            expectations.pop(0)

                except Empty:

                    pass

        except ReadTimeout as timeout_exception:

            print(  # noqa: T201

                "\nFailed to get an expected response from the server within "

                f"{timeout_exception.seconds_passed} seconds."

            )

            received_lines = "".join(received_input)

            print(f"\n--- Received input ---\n\n{received_lines}")  # noqa: T201

>           raise ReadTimeout(timeout_exception.seconds_passed) from None

E           tests.end2end.server.ReadTimeout: 30.82



tests\end2end\server.py:336: ReadTimeout

----------- Latest Logs dir: D:\a\strictdoc\strictdoc\latest_logs\ ------------

=========================== short test summary info ===========================

FAILED tests/end2end/project_options/server_host/02_option_specified/test_02_server_host_specified.py::Test_ProjectOptions_ServerHostAndPort_02_OptionSpecified::test - tests.end2end.server.ReadTimeout: 30.82
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -105,3 +105,9 @@ Changelog = "https://github.com/strictdoc-project/strictdoc/blob/main/CHANGELOG.
 # Funding = "https://..."
 Homepage = "https://strictdoc.readthedocs.io/en/stable/"
 Source = "https://github.com/strictdoc-project/strictdoc"
+
+[tool.pytest.ini_options]
+addopts = "--import-mode=importlib"
+pythonpath = [
+  "."
+]

--- a/tests/end2end/project_options/project_title/01_option_not_specified/test_case.py
+++ b/tests/end2end/project_options/project_title/01_option_not_specified/test_case.py
@@ -9,7 +9,7 @@ from tests.end2end.server import SDocTestServer
 path_to_this_test_file_folder = os.path.dirname(os.path.abspath(__file__))
 
 
-class Test_ProjectOptions_ProjectTitle_02_OptionSpecified(E2ECase):
+class Test(E2ECase):
     def test(self):
         with SDocTestServer(
             input_path=path_to_this_test_file_folder
@@ -18,6 +18,4 @@ class Test_ProjectOptions_ProjectTitle_02_OptionSpecified(E2ECase):
 
             screen_project_index = Screen_ProjectIndex(self)
             screen_project_index.assert_on_screen()
-            screen_project_index.assert_header_project_name(
-                "Test Project title read from TOML file"
-            )
+            screen_project_index.assert_header_project_name("Untitled Project")

--- a/tests/end2end/project_options/project_title/02_option_specified/test_case.py
+++ b/tests/end2end/project_options/project_title/02_option_specified/test_case.py
@@ -9,7 +9,7 @@ from tests.end2end.server import SDocTestServer
 path_to_this_test_file_folder = os.path.dirname(os.path.abspath(__file__))
 
 
-class Test_ProjectOptions_ProjectTitle_01_OptionNotSpecified(E2ECase):
+class Test(E2ECase):
     def test(self):
         with SDocTestServer(
             input_path=path_to_this_test_file_folder
@@ -18,4 +18,6 @@ class Test_ProjectOptions_ProjectTitle_01_OptionNotSpecified(E2ECase):
 
             screen_project_index = Screen_ProjectIndex(self)
             screen_project_index.assert_on_screen()
-            screen_project_index.assert_header_project_name("Untitled Project")
+            screen_project_index.assert_header_project_name(
+                "Test Project title read from TOML file"
+            )

--- a/tests/end2end/project_options/server_host/01_option_not_specified/test_case.py
+++ b/tests/end2end/project_options/server_host/01_option_not_specified/test_case.py
@@ -9,22 +9,23 @@ from tests.end2end.server import SDocTestServer
 path_to_this_test_file_folder = os.path.dirname(os.path.abspath(__file__))
 
 
-class Test_ProjectOptions_ServerHostAndPort_02_OptionSpecified(E2ECase):
+class Test(E2ECase):
     def test(self):
+        # Running this test can eventually cause conflicts when running
+        # tests in parallel. Hoping that testing the server_host option is the
+        # only case where we have to customize ports.
         custom_server_stderr_expectations = [
             "INFO:     Application startup complete.",
-            "INFO:     Uvicorn running on http://localhost:51000 (Press CTRL+C to quit)",  # noqa: E501
+            "INFO:     Uvicorn running on http://127.0.0.1:5112 (Press CTRL+C to quit)",  # noqa: E501
         ]
 
         with SDocTestServer(
             input_path=path_to_this_test_file_folder,
             expectations=custom_server_stderr_expectations,
-            port=51000,
+            port=5112,
         ) as test_server:
             self.open(test_server.get_host_and_port())
 
             screen_project_index = Screen_ProjectIndex(self)
             screen_project_index.assert_on_screen()
-            screen_project_index.assert_header_project_name(
-                'Test project with a host "localhost" and a port 51000'
-            )
+            screen_project_index.assert_header_project_name("Untitled Project")

--- a/tests/end2end/project_options/server_host/02_option_specified/test_case.py
+++ b/tests/end2end/project_options/server_host/02_option_specified/test_case.py
@@ -9,23 +9,22 @@ from tests.end2end.server import SDocTestServer
 path_to_this_test_file_folder = os.path.dirname(os.path.abspath(__file__))
 
 
-class Test_ProjectOptions_ServerHostAndPort_01_OptionNotSpecified(E2ECase):
+class Test(E2ECase):
     def test(self):
-        # Running this test can eventually cause conflicts when running
-        # tests in parallel. Hoping that testing the server_host option is the
-        # only case where we have to customize ports.
         custom_server_stderr_expectations = [
             "INFO:     Application startup complete.",
-            "INFO:     Uvicorn running on http://127.0.0.1:5112 (Press CTRL+C to quit)",  # noqa: E501
+            "INFO:     Uvicorn running on http://localhost:5123 (Press CTRL+C to quit)",  # noqa: E501
         ]
 
         with SDocTestServer(
             input_path=path_to_this_test_file_folder,
             expectations=custom_server_stderr_expectations,
-            port=5112,
+            port=5123,
         ) as test_server:
             self.open(test_server.get_host_and_port())
 
             screen_project_index = Screen_ProjectIndex(self)
             screen_project_index.assert_on_screen()
-            screen_project_index.assert_header_project_name("Untitled Project")
+            screen_project_index.assert_header_project_name(
+                'Test project with a host "localhost" and a port 51000'
+            )

